x-pack/plugins/notifications/server/services/types.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export interface EmailService {
+  sendPlainTextEmail(payload: PlainTextEmail): Promise<void>;
+}
+
```

Comment:
> I wonder if we couldn't find a better way to handle the case where the email service couldn't be created (for any reason). Just having the email property on the contract be undefined feels kinda wrong to me...

@pgayvallet @cnasikas  The `start()` contract of the plugin has been modified and it now offers a couple of methods, to check for availability, and to actually get the email service (see below).

Note that the later will throw an Exception if the service is not available (due to missing dependencies or missing configuration). This way, we have a way to inform consumers as to why the service is not available.